ivy/functional/frontends/torch/tensor.py

```diff
@@ -8,10 +8,15 @@
 
 
 class Tensor:
-    def __init__(self, array, device=None):
-        self._ivy_array = ivy.asarray(
-            array, dtype=torch_frontend.float32, device=device
-        )
+    def __init__(self, array, device=None, _init_overload=False):
+
+        if _init_overload:
+            self._ivy_array = ivy.asarray(array, device=device)
```

Comment:
Have this be `        self._ivy_array = (
            ivy.array(array) if not isinstance(array, ivy.Array) else array
        )` like with `EagerTensor` and `DeviceArray`

Reply:
right!